rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -55,252 +49,116 @@ class Airdrop(NamedTuple):
-        'https://poap.delivery/muir-glacier',
-        'Muir Glacier',
-    ),
-    'proof_of_gucci': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/43b2c4bb73923d7bb3eaf3b329f7f7a1/raw/e5adb753a3a86f0a50bf93137e2c4adc61548293/poap_proof_of_gucci.json',
-        'https://poap.delivery/proof-of-gucci',
-        'YFI - Proof of Gucci',
-    ),
-    'proof_of_gucci_design_competition': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/40c6b2a94d4b8d7f442522b099e5e258/raw/a0851ce92ca2f668e17a97c9df982bc3e12f9bb3/poap_proof_of_gucci_design_competition.json',
-        'https://poap.delivery/proof-of-gucci-design',
-        'YFI - Proof of Gucci Design Competition',
-    ),
-    'resuscitators': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/0ac0216f82f16453b74a40529384a152/raw/f003083090efd834bcee1d0d1fe4e218380aa0cf/poap_resucitators.json',
-        'https://poap.delivery/medalla-resuscitator',
-        'Medalla Resuscitators',
-    ),
-    'yam': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/d676e1f3db8df96928c2501c6be434ac/raw/fa0a603117f6e3e807a49491924aaaa2bc89179a/poap_yam.json',
-        'https://poap.delivery/yam',
-        'Yam Heroes',
-    ),
-    'ycover': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/727c05f3a9cab79059258528595c102e/raw/59ff367bd532f632ce94e69900709155a55de82a/poap_ycover.json',
-        'https://poap.delivery/ycover',
-        'A New Face For yCover',
-    ),
-    'yfi_og': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/58862ec2b398c770d60c24e50e18e50c/raw/3292995f530baedcdacde02526c6b2bd1de0e110/poap_yfi_og.json',
-        'https://poap.delivery/yfi-og',
-        'I Played 4 YFI',
-    ),
-}
-
-
-def get_airdrop_data(name: str, data_dir: Path) -> Iterator[list[str]]:
-    airdrops_dir = data_dir / 'airdrops'
+def _parse_airdrops(database: 'DBHandler', airdrops_data: dict[str, Any]) -> dict[str, Airdrop]:
+    """Parses the airdrops' data from airdrops metadata index. Also, creates the new token if
+    it's not present in the DB.
+
+    May raise - RemoteError if the metadata is invalid.
+    """
+    airdrops: dict[str, Airdrop] = {}
+    for protocol_name, airdrop_data in airdrops_data.items():
+        try:
+            if 'new_asset_data' in airdrop_data:
+                try:
+                    if (chain_id_raw := airdrop_data['new_asset_data'].get('chain_id')) is not None:  # noqa: E501
+                        chain_id = ChainID.deserialize(chain_id_raw)
+                    new_asset_type = AssetType.deserialize(airdrop_data['new_asset_data']['asset_type'])  # noqa: E501
+                except DeserializationError as e:
+                    raise RemoteError(f'Airdrops Index contains an invalid ChainID or AssetType of a token for {protocol_name}. {e!s}') from e  # noqa: E501
```

Comment:
In case of error processign the information wouldn't it be better to log and continue instead of failing?